### DISCUSSION TITLE
Template for nested relations

### DISCUSSION THREAD

**Discussion by joselcvarela** [directus/directus#5165] [Feature Requests]
I was creating a view with cards and when choosing the title from inside translations field the view shows "--"

I have made some debug and found this: https://github.com/directus/directus/blob/6763908070aba9d281c56b449ad230315581116f/app/src/views/private/components/render-template/render-template.vue#L64-L66

So the template is defined this way ```{{translations.title}}```. But translations is an array, so it's not possible to access it.

I have opened this here in the discussion because I don't know what to expect. I think we have two options:
- Select the first relation (in this case translation) and that would be super easy for us, but not so much flexible for the user
- Let the user choose another field to match and use that relation. For example, if I have EN and PT languages I want to show from PT language. This alternative would be more pleasant for the user, but will be more complex to implement.

Here are some screenshots:
![image](https://user-images.githubusercontent.com/14039341/115426556-85949280-a1f8-11eb-8978-87d86a54547d.png)

Let me hear your thoughts

**Comment by rijkvanzanten** [comment]
> because I don't know what to expect.

Same here! A nested collection set (eg o2m/m2m/translations/etc) is inherently a list of things, so it's a little difficult to infer which one of those things to show, or whether or not to show them all 🤔 We do currently have the related-values display that will render `13 Items` with a dropdown preview, but that's not exactly what you're referring to here of course. 

I think the only real way is to have a display that can be configured with some sort of filter to fetch the single item within the nested data set to render, that way you can set "give me language pt-BR" or "related article with most upvotes" 🤔

**Reply by joselcvarela** [reply]
Just realized that this also happens for templates on "related-values" interface. For example, for One To Many (O2M) and Many To One (M2O) relations. 

(added full name on relations for better SEO 🙂)

**Reply by SeanDylanGoff** [reply]
There is a proposal to solve this (and more) in a flexible way. Till then you can use #3592 for a rough solution.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
